webapp/src/Controller/API/SubmissionController.php

```diff
@@ -333,6 +333,10 @@ public function addSubmissionAction(Request $request): string
 
         if ($submissionId = $request->request->get('id')) {
             if ($this->isGranted('ROLE_API_WRITER')) {
+                if (preg_match(DOMJudgeService::EXTERNAL_IDENTIFIER_REGEX, $submissionId) !== 1) {
```

Comment:
Do we actually care? As in: can we allow more than in the spec?

Reply:
Well we expose them through the API as their ID. Personally I don’t care and we barely use them, but I can imagine people using the api assuming they are safe to store as file names for example. Which is exactly what the CDS does

Reply:
I wouldn't hard fail on something that is irrelevant to us.

Reply:
I can remove this check if you want. But what about setting external ids of other entities? If you used the UI we already didn’t allow it before. Do you want to drop that?

Reply:
That's a fair point. Feel free to submit as is.